c/tskit/trees.c

```diff
@@ -4685,3 +4751,255 @@ tsk_tree_kc_distance(tsk_tree_t *self, tsk_tree_t *other, double lambda, double
     }
     return ret;
 }
+
+static int
+check_kc_distance_tree_sequence_inputs(tsk_treeseq_t *self, tsk_treeseq_t *other)
+{
+    tsk_tree_t t1, t2;
+    int ret = 0;
+
+    memset(&t1, 0, sizeof(tsk_tree_t));
+    memset(&t2, 0, sizeof(tsk_tree_t));
+
+    if (self->tables->sequence_length != other->tables->sequence_length) {
+        ret = TSK_ERR_SEQUENCE_LENGTH_MISMATCH;
+        goto out;
+    }
+
+    ret = check_kc_distance_samples_inputs(self, other);
+    if (ret != 0) {
+        goto out;
+    }
+
+    ret = tsk_tree_init(&t1, self, TSK_SAMPLE_LISTS);
+    if (ret != 0) {
+        goto out;
+    }
+    ret = tsk_tree_init(&t2, other, TSK_SAMPLE_LISTS);
+    if (ret != 0) {
+        goto out;
+    }
+
+    ret = tsk_tree_first(&t1);
+    if (ret != 1) {
+        goto out;
+    }
```

Comment:
Ah, OK, I see what you've done here. Let's not do a full tree generation pass just to check the inputs. We can easily run ``check_kc_distance_tree_inputs(&t1, &t2)`` as part of the main loop below. Generating the trees is an expensive operation.